cpp/dolfinx/fem/assemble_matrix_impl.cpp

```diff
@@ -132,6 +132,14 @@ void fem::impl::assemble_cells(
   Eigen::Matrix<PetscScalar, Eigen::Dynamic, Eigen::Dynamic, Eigen::RowMajor>
       Ae;
 
+  const Eigen::Ref<const Eigen::Array<bool, Eigen::Dynamic, Eigen::Dynamic>>
```

Comment:
I am not sure here, maybe @garth-wells knows, but I would add Eigen::RowMajor also here (and all below), to avoid any possible magic performed by Eigen when trying to proxy RowMajor as ColumnMajor. Remember, default Eigen storage is ColumnMajor.

Reply:
Of course, don't do it for 1D arrays.

Reply:
Good point, I've added them throughout